tests/framework/hooks/test_hook.py

```diff
@@ -1,30 +1,97 @@
+import pytest
 from kedro.io import DataCatalog
+from kedro.io.memory_dataset import MemoryDataset
 from kedro.pipeline import node
 from kedro_datasets.pandas import CSVDataset
+from pandera.errors import SchemaError
 from pandera.io import from_yaml
 
 from kedro_pandera.framework.hooks.pandera_hook import PanderaHook
 
 
-def test_hook():
-    test_schema = from_yaml("tests/data/iris_schema.yml")
+def _get_test_catalog(csv_file, schema_file, convert=False):
+    test_schema = from_yaml(schema_file)
     test_catalog = DataCatalog(
         {
             "iris": CSVDataset(
-                filepath="tests/data/iris.csv",
-                metadata={"pandera": {"schema": test_schema}},
-            )
+                filepath=csv_file,
+                metadata={"pandera": {"schema": test_schema, "convert": convert}},
```

Comment:
We don't need the convert key in this PR, this is moved to another one

Reply:
That was indeed from the other PR. Removed.